### ISSUE TITLE
`get_data` return value  depends on complexity of file for tsserver

### ISSUE THREAD

**Issue by yannsartori** [SmiteshP/nvim-navic#24]
Hello,
I recently tried to port over from `nvim-gps` to `nvim-navic`. One thing I noticed which is interesting is that `get_data` seems to be dependent on the complexity of the file. For instance, if I have the following file:

```typescript
class Test {
  private a: boolean = false;
  private b: boolean = false;

  constructor(a: boolean, b: boolean) {
    this.a = a;
    this.b = b;
  }

  foo() {
    return 1;
  }
}
```
and I call `get_data` on the `a` attribute, I get the following response:
```
{ {
    icon = " ",
    kind = 5,
    name = "Test",
    type = "Class"
  }, {
    icon = " ",
    kind = 7,
    name = "a",
    type = "Property"
  } }
```
which makes sense. However, if I add a function to get the following file:
```typescript
class Test {
  private a: boolean = false;
  private b: boolean = false;

  constructor(a: boolean, b: boolean) {
    this.a = a;
    this.b = b;
  }

  foo() {
    return 1;
  }

  bar() {
    return 2;
  }
}
```

Doing the same on `a` yields the following response:

```
{ {
    icon = " ",
    kind = 5,
    name = "Test",
    type = "Class"
  } }
```
If I add `private c: boolean = false;`, I get the above (missing the property) response as above. However, if I then add the following line, making my file this:

```typescript
class Test {
  private a: boolean = false;
  private b: boolean = false;
  private c: boolean = false;
  private d: boolean = false;

  constructor(a: boolean, b: boolean) {
    this.a = a;
    this.b = b;
  }

  foo() {
    return 1;
  }

  bar() {
    return 2;
  }
}
```

I get back

```
{ {
    icon = " ",
    kind = 5,
    name = "Test",
    type = "Class"
  }, {
    icon = " ",
    kind = 7,
    name = "a",
    type = "Property"
  } }
```

I also use python in my daily workflow, and haven't noticed similar issues there, so I think it might be LSP dependent?

This is also with just tsserver running-- I don't have `null-ls` or anything else.

This is my lsp `on_attach` function; nothing special here:
```lua
  if client.name == "tsserver" then
    client.server_capabilities.renameProvider = false
    client.server_capabilities.referencesProvider = false
  end
  
  lsp_keymaps(client, bufnr)
  lsp_highlight_document(client)

  if client.server_capabilities.documentSymbolProvider then
    require("nvim-navic").attach(client, bufnr)
  end
```

**Comment by yannsartori** [comment]
I think this is not an inherent issue with `tsserver` though. If I run `vim.lsp.buf_request_all` manually, I get this output:
```lua
result = { {
        children = { {
            children = {},
            detail = "",
            kind = 9,
            name = "constructor",
            range = {
              end = {
                character = 3,
                line = 7
              },
              start = {
                character = 2,
                line = 4
              }
            },
            selectionRange = {
              end = {
                character = 3,
                line = 7
              },
              start = {
                character = 2,
                line = 4
              }
            }
          }, {
            children = {},
            detail = "",
            kind = 7,
            name = "a",
            range = {
              end = {
                character = 29,
                line = 1
              },
              start = {
                character = 2,
                line = 1
              }
            },
            selectionRange = {
              end = {
                character = 11,
                line = 1
              },
              start = {
                character = 10,
                line = 1
              }
            }
          }, {
            children = {},
            detail = "",
            kind = 7,
            name = "b",
            range = {
              end = {
                character = 29,
                line = 2
              },
              start = {
                character = 2,
                line = 2
              }
            },
            selectionRange = {
              end = {
                character = 11,
                line = 2
              },
              start = {
                character = 10,
                line = 2
              }
            }
          }, {
            children = {},
            detail = "",
            kind = 6,
            name = "bar",
            range = {
              end = {
                character = 3,
                line = 15
              },
              start = {
                character = 2,
                line = 13
              }
            },
            selectionRange = {
              end = {
              character = 5,
                line = 13
              },
              start = {
                character = 2,
                line = 13
              }
            }
          }, {
            children = {},
            detail = "",
            kind = 6,
            name = "foo",
            range = {
              end = {
                character = 3,
                line = 11
              },
              start = {
                character = 2,
                line = 9
              }
            },
            selectionRange = {
              end = {
                character = 5,
                line = 9
              },
              start = {
                character = 2,
                line = 9
              }
            }
          } },
        detail = "",
        kind = 5,
        name = "Test",
        range = {
          end = {
            character = 1,
            line = 16
          },
          start = {
            character = 0,
            line = 0
          }
        },
        selectionRange = {
          end = {
            character = 10,
            line = 0
          },
          start = {
            character = 6,
            line = 0
          }
        }
      } }
  }
  ```
  which has all the symbols, namely `a`.

So I think it has to do how, given the symbols, the code context is reconstituted

**Comment by SmiteshP** [comment]
Should be fixed by PR #23, Can you update and check if this is fixed?

**Comment by SmiteshP** [comment]
>Sorry for the noise, I saw that PR was opened before this ticket, but didn't realize that this was the issue!

No worries! :)
